core/src/main/java/org/apache/accumulo/core/crypto/CryptoServiceFactory.java

```diff
@@ -25,11 +26,38 @@
 
   /**
    * Create a new crypto service configured in {@link Property#INSTANCE_CRYPTO_SERVICE}.
+   *
+   * @param useAccumuloClassloader
+   *          Determine if Accumulo class loader is used. When calling from accumulo server code,
+   *          pass true. When calling from Accumulo client code, pass false.
    */
-  public static CryptoService newInstance(AccumuloConfiguration conf) {
-    CryptoService newCryptoService = Property.createInstanceFromPropertyName(conf,
-        Property.INSTANCE_CRYPTO_SERVICE, CryptoService.class, new NoCryptoService());
+  public static CryptoService newInstance(AccumuloConfiguration conf,
+      boolean useAccumuloClassloader) {
```

Comment:
Could make the parameter an enum and then set useAccumuloClassloader to the appropriate value.  I was thinking something like Caller.CLIENT or Scope.CLIENT, similar to what we did with Scope in CryptoEnvironment.  There is some dislike for booleans as parameters.  I found myself having to go back and check the javadoc that false = client.

Reply:
This would make reading the code easier.

Reply:
done in 455f2ca